### ISSUE TITLE
Relative paths seem to not be supported

### ISSUE THREAD

**Issue by mosuem** [bmw-tech/dart_apitool#198]
**Describe the bug**
While trying out the PR health workflow in dart-lang/native, I saw that it throws for some of the packages in that monorepo.

**To Reproduce**
```
#git clone dart-lang/native
cd native/
flutter pub global run dart_apitool:main diff --no-check-sdk-version --old pub://ffigen --new pkgs/ffigen --report-format json --report-file-path report.json
```
**Actual and Expected behavior**
Throws an error
```
Error: Exception: Because ffigen depends on objective_c from path which doesn't exist (could not find package objective_c at "../objective_c"), version solving failed.
```
But this package does exist there. Probably some copying doesn't copy the path dependencies?

**System info (please complete the following information):**
 - Output of `dart --version`: `Dart SDK version: 3.7.0-207.0.dev (dev) (Mon Dec 2 12:02:22 2024 -0800) on "linux_x64"`
 - dart_apitool version (`dart-apitool --version`): `0.20.0`

**Additional context**
-

**Comment by devmil** [comment]
@mosuem thanks for reporting this
The issue is that ffigen on pub contains dependency overrides and dart_apitool is trying to execute pub get on a downloaded copy which does not work.
The linked PR removes any dependency overrides in that situation which should be no issue as those overrides only affect local pub gets anyways

**Comment by mosuem** [comment]
Thanks! That will work well for packages which are ready to be published. However, this will not help for packages during during development, where multiple packages may be touched at the same time. Should I open a new issue for this? This should be less common though.

**Comment by devmil** [comment]
@mosuem good point. Will test that but I think this works already as we execute pub get in place and transfer the resulting package config while turning relative paths into absolute ones.

**Comment by devmil** [comment]
@mosuem sorry for the delay. The fix is released now (0.20.1) and I added an integration test that covers ffigen (so a package in a mono repo with relative references from source) [here](https://github.com/bmw-tech/dart_apitool/pull/200)
